### ISSUE TITLE
As BrennerLEC technical expert I would like that the Open Data Hub can manage "physical" and "virtual" station locations associated to the low-cost air quality sensors used

### ISSUE THREAD

**Issue by rcavaliere** [noi-techpark/opendatahub-collectors#37]
We need to handle a particular case for the dataset air quality provided by A22 through their AUGE platform, supplied by algorab.

The low-cost sensors in use will be periodically disinstalled from their physical locations on the highway, brought to an intercalibration site in Trento, and then installed again on a physical location on the highway that could be different from the previous one.

What we need is to have a modeling of the physical locations on the highway, including the intercalibration site, and an association with the sensor (identified by the identifier AQxx) that is currently installed there. This should ideally be organized through this mapping table: https://github.com/noi-techpark/bdp-commons/blob/main/data-collectors/environment-a22/src/main/resources/mappings/stationMappings.csv

What we have to ensure is:

- physical stations are always visible on the Open Data Hub
- physical stations always provide the detail of the associated sensor which is currently installed
- physical stations always provide the detail of the history about which sensor was installed when on that physical location (this should be possible with the metadata history)

**Comment by rcavaliere** [comment]
@clezag you mean the physical stations? Actually they should all be already available in the DB, maybe for some stations we have to check if it is available.

**Comment by clezag** [comment]
@rcavaliere since the old data model has the sensor ID as the main station identifier, we only have the position of currently mapped and active stations/sensors. But I can extract the gps points from the old CSV file history if you don't have them at hand.

For the stations named `_bis`, `4metri`, `centrale` or some combination thereof, what do these indicate?

**Comment by clezag** [comment]
@rcavaliere It's the usual issue of analytics not indicating inactive stations in any way.  
This is an inactive one from the previous implementation that's remained because the station code changed.  
There is one right next to it, with no data, that has the correct history.  
No data is also correct, because right now there is no sensor attached to this station.  

You can check the active histories here:
https://mobility.api.opendatahub.testingmachine.eu/v2/flat/EnvironmentStation?where=sorigin.eq.a22-algorab,sactive.eq.true

I will delete these "noisy" inactive stations, and re-import all the raw data so that the new stations are populated. 

In the meantime you can proceed with testing/checking, just make sure you only look at active stations i.e. the ones matching their station code to the 2nd column in the CSV.

**Comment by rcavaliere** [comment]
@clezag let's leave like this. In these periods it could have been that the sensors were switched on without being installed at a station... I think that in case we can always correct the historical association between stations and sensors, or?

**Comment by rcavaliere** [comment]
@clezag yes clear for me. Let's leave like this at the moment, let's have a look at the result when the elaborations are completed.

**Comment by clezag** [comment]
@rcavaliere looks like everything went well, stations are well populated

**Comment by rcavaliere** [comment]
@clezag thanks for noticing this. The non linear calibration formulas use unfortunately `RH` as input parameter, so I would suggest to remake the computation if we want to be sure everything is fine.

**Comment by rcavaliere** [comment]
@clezag now that I have access to analytics testing I have checked this. Everything seems fine. Before going in production, I would like to make a tests with the latest movements of the sensors, which are not documented yet, so that we test the update position process before going in production and we can then import in production the current positioning situation.

**Comment by rcavaliere** [comment]
@clezag the sensors have been now all positioned, I have changed the CSV file and made this PR but did not work (the error message seems unclear to me, can you tell me what went wrong for the next updates). I upload the file here, in case can you please update it? Let me then check that the sensors are associated in the proper way and we can then start to put everything in production

[stations.csv](https://github.com/user-attachments/files/22233925/stations.csv)

**Comment by clezag** [comment]
@rcavaliere You did everything correctly, there was a unit test failing because it relied on some stations being mapped a certain way.  
I've updated the test so that it uses a copy of the stations.csv, this way the next time this shouldn't happen again.

The new CSV should be up now

**Comment by rcavaliere** [comment]
> The historical data is still there, in the old format, with the stations inactive. 

I am not sure about this statement, some BrennerLEC partners already contacted me where they can find now historical measurements. I can also not see them associated to the inactive stations.

**Comment by clezag** [comment]
@rcavaliere fixed and deployed in production
